### ISSUE TITLE
Is this still working? (Path Issues)

### ISSUE THREAD

**Issue by fredlemieux** [g1eny0ung/Alfred-Colors-workflow#3]
Tried this out on Mac M1, with Alfred v4 and nothing happens when I use it....
![Kapture 2022-03-01 at 22 20 24](https://user-images.githubusercontent.com/7422952/156258649-701de49f-c976-467f-931b-f7ac924c6f99.gif)

**Comment by g1eny0ung** [comment]
Can I see the debug logs? This problem is most likely related to the `PATH`.

**Comment by g1eny0ung** [comment]
See also #2.

**Comment by samplereality** [comment]
I'm also having this problem. This is the error in the debug log:

```Code 1: <internal:/opt/homebrew/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- chunky_png (LoadError)
	from <internal:/opt/homebrew/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from colors.rb:4:in `<main>'```
	
And here is the `gem environment`:

```
RubyGems Environment:
  - RUBYGEMS VERSION: 3.4.10
  - RUBY VERSION: 3.2.2 (2023-03-30 patchlevel 53) [arm64-darwin22]
  - INSTALLATION DIRECTORY: /Users/masample/.gem/ruby/3.2.2
  - USER INSTALLATION DIRECTORY: /Users/masample/.gem/ruby/3.2.0
  - RUBY EXECUTABLE: /opt/homebrew/opt/ruby/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /Users/masample/.gem/ruby/3.2.2/bin
  - SPEC CACHE DIRECTORY: /Users/masample/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /opt/homebrew/Cellar/ruby/3.2.2_1/etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - arm64-darwin-22
  - GEM PATHS:
     - /Users/masample/.gem/ruby/3.2.2
     - /Users/masample/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => true
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /Users/masample/.gem/ruby/3.2.2/bin
     - /opt/homebrew/opt/ruby/bin
     - /usr/local/opt/ruby/bin
     - /Users/masample/.gem/ruby/3.2.2/bin
     - /Users/masample/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0/bin
     - /Users/masample/.rubies/ruby-3.2.2/bin
     - /usr/local/opt/ruby/bin
     - /usr/local/lib/ruby/gems/3.1.0/bin
     - /Users/masample/.rbenv/shims
     - /opt/homebrew/bin
     - /opt/homebrew/sbin
     - /usr/local/bin
     - /System/Cryptexes/App/usr/bin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin
     - /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin
     - /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin
     - /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
```
     
I think it might be a PATH problem but I can't figure it out for the life of me! (On an M2)

**Comment by g1eny0ung** [comment]
@samplereality Could you check this location (`- INSTALLATION DIRECTORY: /Users/masample/.gem/ruby/3.2.2`) and see if `chunky_png` is installed? Gem will install gems there. Indeed, this is usually a path problem.

<img width="734" alt="image" src="https://github.com/g1eny0ung/Alfred-Colors-workflow/assets/15034155/b6072b61-543c-4abd-8b9e-c0425760ebd4">

**Comment by samplereality** [comment]
I just checked, and chunky_png is there:

<img width="682" alt="image" src="https://github.com/g1eny0ung/Alfred-Colors-workflow/assets/1093156/c83243e2-1b14-4f47-9c2b-18062f81dfe6">

**Comment by g1eny0ung** [comment]
Perhaps I know why. 🤔 The problem may be related to `$GEM_PATH`. I get these paths from my Mac: 

<img width="536" alt="image" src="https://github.com/g1eny0ung/Alfred-Colors-workflow/assets/15034155/2ea704a0-aad7-48e5-a0dd-ad62717671b5">

Could you try appending `/opt/homebrew/Cellar/ruby/3.2.2_1/lib/ruby/gems/3.2.0` to your gem paths and see if it works:

```
export GEM_PATH=$(gem env path):/opt/homebrew/Cellar/ruby/3.2.2_1/lib/ruby/gems/3.2.0
```

**Comment by samplereality** [comment]
No, that didn't work either. Still not sure what's up. But thanks for trying!

**Comment by g1eny0ung** [comment]
> No, that didn't work either. Still not sure what's up. But thanks for trying!

You're welcome. However, I still believe that the issue is likely related to the path. I hope you're able to resolve it and share the solution here. ❤️

**Comment by hyunjeen** [comment]
This is exactly the problem with Gem Path, and here's how I solved it 

0. gem uninstall chunky_png
1.)touch ~/.gemrc
2) vim ~/.gemrc

> Gem environment
Gem Path: 
/Users/~/~/~ Copy the address in the path 

3) Paste the copied text inside the .gemrc file and hit
Reopen the terminal 
gem install chunky_png

run alfred command c #ffffff 
working !

This worked for me and I hope it works for others.

Translated with www.DeepL.com/Translator (free version)

**Comment by stdavis** [comment]
Manually updating the script filter for this workflow to remove the paths to the ruby executable solved this issue for me:
![image](https://github.com/user-attachments/assets/eacb2260-36cd-46d1-a7c5-5466b06f25dc)

**Comment by maymaymaybe8** [comment]
Open Alfred Preferences → Workflows → (Colors) → **Debug** and run the keyword once. If you see **Code 127: /bin/bash: line 3: /opt/homebrew/opt/ruby/bin/ruby: No such file or directory**, this workflow can’t find the Homebrew Ruby it expects at that path. Install Ruby via Homebrew (and ensure the chunky_png gem is installed for that Ruby).

**Comment by g1eny0ung** [comment]
> Open Alfred Preferences → Workflows → (Colors) → **Debug** and run the keyword once. If you see **Code 127: /bin/bash: line 3: /opt/homebrew/opt/ruby/bin/ruby: No such file or directory**, this workflow can’t find the Homebrew Ruby it expects at that path. Install Ruby via Homebrew (and ensure the chunky_png gem is installed for that Ruby).

@maymaymaybe8 I've updated the command to:

`$(brew --prefix)/opt/ruby/bin/ruby colors.rb "{query}"`

Using `brew --prefix` to find the Homebrew path is the correct approach, rather than manually setting it. You can download it again and try it.

If you still have problems, you can try manually modifying the path to match your environment.

**Comment by maymaymaybe8** [comment]
@g1eny0ung Thank you.
I ran into the same problem as the original poster when I first installed the workflow, but I was able to solve it after asking ChatGPT. After reading this issue, I thought they might be dealing with a similar problem, so I shared my solution here in case it helps. I’m no longer having this issue. Thanks — I really like this workflow and have been using it.

**Comment by g1eny0ung** [comment]
@maymaymaybe8 You're welcome. I'm glad this workflow can help you!
